tests/js/spec/stores/savedSearchesStore.spec.jsx

```diff
@@ -1,28 +1,237 @@
+    await tick();
+
+    expect(SavedSearchesStore.get().savedSearches).toHaveLength(3);
+    expect(SavedSearchesStore.get().savedSearches[0]).toEqual(
+      expect.objectContaining({
+        id: null,
+        isPinned: true,
+        type: 0,
+        query: 'level:info',
+        name: 'My Pinned Search',
+      })
+    );
+  });
+
+  it('changes pinned search from a custom search to an existing search', async function() {
+    const searches = TestStubs.Searches();
+
+    Client.addMockResponse({
+      url: '/organizations/org-1/searches/',
+      body: [
+        {
+          id: null,
+          isPinned: true,
+          isPrivate: true,
+          type: 0,
+          query: 'assigned:me',
+        },
+        ...searches,
+      ],
+    });
+    fetchSavedSearches(api, 'org-1');
+    await tick();
+
+    pinSearch(api, 'org-1', 0, searches[1].query);
+    await tick();
+
+    // Order should remain the same
+    expect(SavedSearchesStore.get().savedSearches[1]).toEqual(
+      expect.objectContaining({
+        id: '1',
+        isPinned: true,
+        isPrivate: false,
+        isGlobal: true,
+        type: 0,
+        name: 'Unresolved Issues',
+        query: 'is:unresolved',
+      })
+    );
+
+    // Saved custom search should be removed
+    expect(SavedSearchesStore.get().savedSearches).toHaveLength(2);
+  });
+
+  it('changes pinned search from an existing search to another existing search', async function() {
+    const searches = TestStubs.Searches();
+
+    Client.addMockResponse({
+      url: '/organizations/org-1/searches/',
+      body: [{...searches[0], isPinned: true}, searches[1]],
+    });
+    fetchSavedSearches(api, 'org-1');
+    await tick();
+
+    pinSearch(api, 'org-1', 0, searches[1].query);
+    await tick();
+
+    expect(SavedSearchesStore.get().savedSearches).toHaveLength(2);
+
+    expect(SavedSearchesStore.get().savedSearches[0]).toEqual(
+      expect.objectContaining({
+        id: '2',
+        isPinned: false,
+        isPrivate: false,
+        isGlobal: true,
+        type: 0,
+        name: 'Needs Triage',
+        query: 'is:unresolved is:unassigned',
+      })
+    );
+
+    expect(SavedSearchesStore.get().savedSearches[1]).toEqual(
+      expect.objectContaining({
+        id: '1',
+        isPinned: true,
+        isPrivate: false,
+        isGlobal: true,
+        type: 0,
+        name: 'Unresolved Issues',
+        query: 'is:unresolved',
+      })
+    );
+  });
+
+  it('unpins a user custom search (private)', async function() {
+    const searches = TestStubs.Searches();
+
+    Client.addMockResponse({
+      url: '/organizations/org-1/searches/',
+      body: [
+        {
+          id: null,
+          isPinned: true,
+          isPrivate: true,
+          type: 0,
+          query: 'assigned:me',
+        },
+        ...searches,
+      ],
+    });
+    fetchSavedSearches(api, 'org-1');
+    await tick();
+
+    unpinSearch(api, 'org-1', 0, searches[0]);
```

Comment:
Will our client-side state manipulation handle the scenario where a user un-pins a search that is also an organization saved search?

Reply:
Yeah, there's a test for that: https://github.com/getsentry/sentry/pull/12737/files/553e943638a1dd1534d847d69b18a185d4d53d67#diff-cb2c47c4efc1613f2aacb3629fb9c899R198

Reply:
Ah nm, I have my flags all mixed up.

Reply:
Fixed up the flags and updated tests https://github.com/getsentry/sentry/pull/12737/commits/813730035392941ec9585ea77ffa10a6b0ba3e5f